src/server/worker/pipeline/common.go

```diff
@@ -19,20 +20,87 @@ import (
 	"github.com/pachyderm/pachyderm/src/server/worker/common"
 	"github.com/pachyderm/pachyderm/src/server/worker/driver"
 	"github.com/pachyderm/pachyderm/src/server/worker/logs"
+
+	"github.com/rjeczalik/notify"
 )
 
-func openAndWait() error {
-	// at the end of file, we open the pipe again, since this blocks until something is written to the pipe
-	openAndWait, err := os.Open("/pfs/out")
-	if err != nil {
-		return err
+// TarFile holder structure
+type TarFile struct {
+	hdr  *tar.Header
+	body []byte
+}
+
+var fileList []TarFile
```

Comment:
I had added the global variables for a quick implementation. As the implementation evolved, I realized we don't need the global state. This should be easy to fix.

Reply:
ah yeah, that makes sense: i thought it looked weird but figured you had had a reason to do it that way